signalfx-tracing/signalfx-nodejs-tracing/mysql/deedScheduler/client.js

```diff
@@ -0,0 +1,99 @@
+// Here we import the tracer before the request library
+// to ensure that its http dependency is instrumented
+const tracer = require('./tracer');
+
+
+const http = require('http');
+const {serverUrl} = require('./config');
+const deedSchedulerUrl = `${serverUrl}/deedScheduler`;
+
+
+function resolveData(res, resolve) {
+  // resolves a Promise with parsed http response
+  let data = '';
+  res.on('data', (d) => {
+    data += d;
+  });
+  res.on('end', () => {
+    resolve(JSON.parse(data));
+  });
+}
+
+
+function addItem(deed, note, day) {
+  return new Promise((resolve, reject) => {
+    const req = http.request(`${deedSchedulerUrl}/add`,
+        {method: 'POST'}, (res) => {
+          resolveData(res, resolve);
+        });
+    req.on('error', (e) => reject(e));
+    const content = {deed, note, day};
+    req.setHeader('Content-Type', 'application/json');
+    req.write(JSON.stringify(content));
+    req.end();
+  });
+}
+
+
+function deleteDeed(deed, day) {
+  return new Promise((resolve, reject) => {
+    const thisDay = day ? day : '__ALL__';
+    const req = http.request(`${deedSchedulerUrl}/deeds/${deed}?day=${thisDay}`,
+        {method: 'DELETE'}, (res) => {
+          resolveData(res, resolve);
+        });
+    req.on('error', (e) => reject(e));
+    req.end();
+  });
+}
+
+
+function listDeeds(day) {
+  return new Promise((resolve, reject) => {
+    const thisDay = day ? day : '__ALL__';
+    const req = http.request(`${deedSchedulerUrl}/deeds?day=${thisDay}`,
+        {method: 'GET'}, (res) => {
+          resolveData(res, resolve);
+        });
+    req.on('error', (e) => reject(e));
+    req.end();
+  });
+}
+
+
+function updateDeed(deed, day, status) {
```

Comment:
s/day/note, including on the backend, or just add note w/ default day?

Reply:
This only updates the status of the deed. 1 for completed. (I didn't include updating notes as this was sufficient to test the 'PUT' endpoint... and that would have increased the arguments to be passed via the command line. A default day may be tricky since the same task can be spread over different days.

Reply:
Switched out note for day in the README